velox/exec/Task.cpp

```diff
@@ -418,6 +422,48 @@ void Task::removeSpillDirectoryIfExists() {
   }
 }
 
+/// Builds the query trace config.
+std::optional<QueryTraceConfig> Task::makeQueryTraceConfig() const {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  const auto queryTraceDir = queryConfig.queryTraceDir();
+  if (queryTraceDir.empty()) {
+    LOG(WARNING) << "Query trace enabled but the trace dir is not set.";
+    return std::nullopt;
+  }
+
+  const auto queryTraceNodes = queryConfig.queryTraceNodes();
+  if (queryTraceNodes.empty()) {
+    LOG(WARNING) << "Query trace enabled but no target nodes.";
+    return std::nullopt;
+  }
+
+  std::vector<std::string> tmp;
+  folly::split(',', queryTraceNodes, tmp);
+  const std::unordered_set<std::string> nodes(
+      std::make_move_iterator(tmp.begin()), std::make_move_iterator(tmp.end()));
+
+  const auto queryTraceWriteBufferSize = queryConfig.queryWriteBufferSize();
+
+  return QueryTraceConfig(nodes, queryTraceDir, queryTraceWriteBufferSize);
+}
+
+std::string Task::createTraceDirectory(const std::string& subDir) const {
+  std::lock_guard<std::mutex> l(traceDirCreateMutex_);
```

Comment:
maybe we should use one lock for both trace and spill directory creation.

Reply:
Got it, done